### ISSUE TITLE
Editor crash when minimizing on Windows (crash in VulkanContext::prepare_buffers)

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#37327]
**Godot version:**
5f1107a

**OS/device including version:**
Windows 10

**Steps to reproduce:**
1. Minimize editor
```
[0] vk_optimusGetInstanceProcAddr
[1] vkGetInstanceProcAddr
[2] VulkanContext::prepare_buffers (x:\inne\godot\source\godot\drivers\vulkan\vulkan_context.cpp:1228)
[3] RenderingDeviceVulkan::prepare_screen_for_drawing (x:\inne\godot\source\godot\drivers\vulkan\rendering_device_vulkan.cpp:6692)
[4] VisualServerViewport::draw_viewports (x:\inne\godot\source\godot\servers\visual\visual_server_viewport.cpp:452)
[5] VisualServerRaster::draw (x:\inne\godot\source\godot\servers\visual\visual_server_raster.cpp:114)
[6] Main::iteration (x:\inne\godot\source\godot\main\main.cpp:2139)
[7] OS_Windows::run (x:\inne\godot\source\godot\platform\windows\os_windows.cpp:661)
[8] widechar_main (x:\inne\godot\source\godot\platform\windows\godot_windows.cpp:162)
[9] _main (x:\inne\godot\source\godot\platform\windows\godot_windows.cpp:186)
[10] main (x:\inne\godot\source\godot\platform\windows\godot_windows.cpp:196)
[11] __scrt_common_main_seh (f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:283)
[12] BaseThreadInitThunk
-- END OF BACKTRACE --
```

**Comment by reduz** [comment]
I am unable to reproduce this, is it still a problem?

**Comment by KoBeWi** [comment]
Yes, just tried with 73262c0

**Comment by MissEmbyr** [comment]
This is windows only I believe and I'm still experiencing it on master

**Comment by marstaik** [comment]
![image](https://user-images.githubusercontent.com/5601015/86884119-f3c3fa80-c0a7-11ea-88eb-c84a204111d2.png)

At first glance it seems that w->swapchain is being set to NULL after the if statement check.
It would seem that there might be some form of thread fighting happening here, but I dont have any experience with the vulkan stack and windows context to know whats going on.

**Comment by marstaik** [comment]
I managed to breakpoint right before the call to fpAcquireNextImageKHR.

![image](https://user-images.githubusercontent.com/5601015/86885067-78fbdf00-c0a9-11ea-957a-d87e29c4e20a.png)

I confirmed that the swapchain is indeed NULL before the call to fpAcquireNextImageKHR.

I set a breakpoint at 1221 in `vulkan_context.cpp`, where the `swapchain` is out of date due to a window resize.
this calls into `_update_swap_chains()`, which at line 839 returns `OK` without having allocated a swapchain due to the window width and height being zero. This causes the next iteration of the do-while to call fpAcquireNextImageKHR with a nullptr.
